src/device/clint.rs

```diff
@@ -1,18 +1,39 @@
-use core::sync::atomic::{AtomicBool, Ordering};
+use core::cmp::min;
+use core::sync::atomic::{AtomicBool, AtomicUsize, Ordering};
 
 use spin::Mutex;
 
-use crate::arch::mie;
+use crate::arch::{mie, Arch, Architecture, Csr};
 use crate::config::PLATFORM_NB_HARTS;
 use crate::device::{DeviceAccess, Width};
 use crate::driver::clint::{
     ClintDriver, MSIP_OFFSET, MSIP_WIDTH, MTIMECMP_OFFSET, MTIMECMP_WIDTH, MTIME_OFFSET,
 };
+use crate::host::MiralisContext;
+use crate::platform::{Plat, Platform};
 use crate::virt::VirtContext;
 use crate::{debug, logger};
 
 // ————————————————————————————— Virtual CLINT —————————————————————————————— //
 
+// We use this structure to avoid false sharing in the virtual clint
+// The typical size of a cache line is 64 bytes
+#[repr(C, align(64))]
+#[derive(Debug)]
+struct TimestampEntry {
+    counter: AtomicUsize,
+    _padding: [u8; 56],
+}
```

Comment:
If you are going for a struct with padding lets maybe add both timestamps (firmware and payload) in the same struct to limit bloating the code size. They are both read by the same core, so that will not add more cache contention while dividing the total timestamp size footprint by 2.